.github/workflows/publish.yaml

```diff
@@ -1,41 +1,90 @@
-# reusable workflow triggered by other actions
+# reusable workflow for publishing all charms in this repo
 name: Publish
 
 on:
   workflow_call:
     secrets:
       charmcraft-credentials:
         required: true
+  workflow_dispatch:
+    inputs:
+      # TODO: Implement this
+      # source_github_branch:
+      #   description: Source branch to publish
+      #   required: false
+      #   default: 'main'
+      destination_channel:
+        description: CharmHub channel to publish to
+        required: false
+        default: 'latest/edge'
 
 jobs:
+  get-charm-paths:
+    name: Generate the Charm Matrix
+    runs-on: ubuntu-latest
+    outputs:
+      charm_paths_list: ${{ steps.get-charm-paths.outputs.CHARM_PATHS_LIST }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Get paths for all charms in repo
+        id: get-charm-paths
+        run: bash .github/workflows/get-charm-paths.sh
+
 
   publish-charm:
     name: Publish Charm
     runs-on: ubuntu-latest
+    needs: get-charm-paths
     strategy:
       fail-fast: false
       matrix:
-        charm:
-          - kfp-api
-          - kfp-persistence
-          - kfp-profile-controller
-          - kfp-schedwf
-          - kfp-ui
-          - kfp-viewer
-          - kfp-viz
+        charm-path: ${{ fromJson(needs.get-charm-paths.outputs.charm_paths_list) }}
```

Comment:
That's great, thanks for implementing this!
Just to keep consistency across all workflows, could you change the matrix of hardcoded charms in `integration.yaml` to use the `get-charm-paths` action too?

Reply:
Giving this a second thought, we should handle that part separately. For example, kfp has integration tests only for 2 charms, so it doesn't make sense to use script in lint and unit but not for integration within the same workflow.